Unix/deprecated/httpclientcxx/httpclientcxx.cpp

```diff
@@ -726,6 +731,22 @@ Result HttpClient::Connect(
     return httpclient::OKAY;
 }
 
+void HttpClient::Cancel( void )
+{
+    pthread_mutex_lock(&s_deleteSemaphoreLock);
```

Comment:
The initialization for the semaphore (and the lock) only happens on connect (line 641 above, in HttpClient::Connect). Will this ONLY be called after HttpClient::Connect() was called? If not, then these things won't be set up, and bad things would happen ...